lib/src/dart_human_crontab_base.dart

```diff
@@ -43,41 +32,79 @@ class HumanCrontab {
         weekday: parts[4]);
   }
 
-  void _validateRange(String val, {int min = 0, required int max}) {
-    if (val == '*') return;
+  static HumanCrontab? tryParse(String value) {
+    try {
+      return HumanCrontab.parse(value);
+    } on CrontabException {
+      return null;
+    }
+  }
 
-    final intval = int.tryParse(val);
-    if (intval != null) {
-      assert(intval >= min && intval <= max,
-          'Invalid crontab string: must be between $min and $max');
-      return;
+  static const _weekdayList = <String>[
+    'Sunday',
+    'Monday',
+    'Tuesday',
+    'Wednesday',
+    'Thursday',
+    'Friday',
+    'Saturday',
+  ];
+
+  static const _monthList = <String>[
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ];
+
+  void _validateRange(String value, {int min = 0, required int max}) {
+    if (value == '*') return;
+
+    final val = int.tryParse(value);
+    if (val != null) {
+      if (val >= min && val <= max) return;
+      throw CrontabException.invalid('must be between $min and $max');
     }
 
-    if (val.contains('/')) {
-      final parts = val.split('/');
-      assert(
-          parts.length == 2, 'Invalid crontab value: must be "*" or a number');
+    if (value.contains('/')) {
+      final parts = value.split('/');
+      if (parts.length != 2) {
+        throw CrontabException.invalid('must be "*" or a number');
```

Comment:
doesnt say what actually was wrong. perhaps include the part that was incorrect?